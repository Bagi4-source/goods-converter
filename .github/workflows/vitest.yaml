name: 'Test'
on:
  pull_request:
  push:

jobs:
  vitest:
    name: Run vitest
    runs-on: ubuntu-latest

    permissions:
      # Required to checkout the code
      contents: read
      # Required to put a comment into the pull-request
      pull-requests: write

    steps:
      - uses: actions/checkout@v2
      - name: 'Install Node'
        uses: actions/setup-node@v2
        with:
          node-version: '18.x'
      - name: 'Install Deps'
        run: npm i
      - name: 'Test'
        run: npm run test
      - name: 'Coverage'
        run: npm run coverage
      - name: 'Report Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2