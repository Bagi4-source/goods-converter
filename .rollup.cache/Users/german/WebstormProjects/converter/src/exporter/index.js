export * from "./exporter";
export * from "./exporter.types";
//# sourceMappingURL=index.js.map