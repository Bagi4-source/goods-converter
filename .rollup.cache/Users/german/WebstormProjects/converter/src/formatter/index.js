export * from "./CSV.formatter";
export * from "./YML.formatter";
export * from "./formater.types";
//# sourceMappingURL=index.js.map