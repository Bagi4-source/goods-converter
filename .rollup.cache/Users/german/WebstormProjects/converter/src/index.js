export * from "./exporter";
export * from "./formatter";
//# sourceMappingURL=index.js.map