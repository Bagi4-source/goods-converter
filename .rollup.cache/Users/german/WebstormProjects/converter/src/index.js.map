{"version":3,"file":"index.js","sourceRoot":"","sources":["index.ts"],"names":[],"mappings":"AAAA,cAAc,YAAY,CAAA;AAC1B,cAAc,aAAa,CAAA","sourcesContent":["export * from \"./exporter\"\nexport * from \"./formatter\""]}