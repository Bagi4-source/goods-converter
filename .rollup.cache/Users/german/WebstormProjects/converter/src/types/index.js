export * from "./Product.types";
export * from "./Category.types";
//# sourceMappingURL=index.js.map