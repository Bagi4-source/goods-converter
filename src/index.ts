export * from "./exporter"
export * from "./formatter"
export * from "./types"