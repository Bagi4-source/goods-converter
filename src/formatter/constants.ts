import {CSVFormatter} from "./CSV.formatter";
import {YMLFormatter} from "./YML.formatter";
import {TildaFormatter} from "./Tilda.formatter";

export const Formatters = {
    CSV: new CSVFormatter(),
    YML: new YMLFormatter(),
    Tilda: new TildaFormatter(),
}
