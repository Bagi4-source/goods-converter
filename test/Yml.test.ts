import { expect, describe, it, vi } from "vitest";

import { Formatters } from "../src";
import { brands, categories, products } from "./constants";

describe("YML formatter", () => {
  const formatter = new Formatters.YMLFormatter();
  vi.useFakeTimers().setSystemTime(new Date("2020-01-01"));

  it("should export YML data", async () => {
    const result = await formatter.format(products, categories, brands, {
      shopName: "Bagi4",
      companyName: "Bagi4",
    });

    expect(result).toMatchSnapshot();
  });
});
