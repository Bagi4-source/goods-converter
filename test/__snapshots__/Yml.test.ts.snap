// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`YML formatter > should export YML data 1`] = `
PassThrough {
  "_events": {
    "close": [
      [Function],
      [Function],
      [Function],
      [Function],
    ],
    "drain": [
      [Function],
      [Function],
      [Function],
    ],
    "error": [
      [Function],
      [Function],
      [Function],
      [Function],
    ],
    "finish": [Function],
    "prefinish": [Function],
    "unpipe": [Function],
  },
  "_eventsCount": 6,
  "_maxListeners": undefined,
  "_readableState": ReadableState {
    "autoDestroy": true,
    "awaitDrainWriters": null,
    "buffer": BufferList {
      "head": null,
      "length": 0,
      "tail": null,
    },
    "closeEmitted": false,
    "closed": false,
    "constructed": true,
    "dataEmitted": false,
    "decoder": null,
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "emittedReadable": false,
    "encoding": null,
    "endEmitted": false,
    "ended": false,
    "errorEmitted": false,
    "errored": null,
    "flowing": null,
    "highWaterMark": 16384,
    "length": 0,
    "multiAwaitDrain": false,
    "needReadable": false,
    "objectMode": false,
    "pipes": [],
    "readableListening": false,
    "reading": false,
    "readingMore": false,
    "resumeScheduled": false,
    "sync": false,
    Symbol(kPaused): null,
  },
  "_writableState": WritableState {
    "afterWriteTickInfo": null,
    "allBuffers": true,
    "allNoop": true,
    "autoDestroy": true,
    "bufferProcessing": false,
    "buffered": [],
    "bufferedIndex": 0,
    "closeEmitted": false,
    "closed": false,
    "constructed": true,
    "corked": 0,
    "decodeStrings": true,
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "ended": false,
    "ending": false,
    "errorEmitted": false,
    "errored": null,
    "finalCalled": false,
    "finished": false,
    "highWaterMark": 16384,
    "length": 0,
    "needDrain": false,
    "objectMode": false,
    "onwrite": [Function],
    "pendingcb": 0,
    "prefinished": false,
    "sync": true,
    "writecb": null,
    "writelen": 0,
    "writing": false,
    Symbol(kOnFinished): [],
  },
  "allowHalfOpen": true,
  Symbol(kCapture): false,
  Symbol(kCallback): null,
}
`;
